File: openapi-generator/dart-jaguar/lib/model/asset_current_locations_response.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

import 'package:openapi/model/asset_cable.dart';
import 'package:openapi/model/asset_current_location.dart';
part 'asset_current_locations_response.jser.dart';

class AssetCurrentLocationsResponse {
   /* The cable connected to the asset */
  @Alias('cable')
  final List<AssetCable> cable;
   /* Engine hours */
  @Alias('engineHours')
  final int engineHours;
   /* Asset ID */
  @Alias('id')
  final int id;
   /* Current location of an asset */
  @Alias('location')
  final List<AssetCurrentLocation> location;
   /* Asset name */
  @Alias('name')
  final String name;
  

  AssetCurrentLocationsResponse(
      

{
     this.cable = const [],  
     this.engineHours = null,  
    
     this.id = null,   this.location = const [],  
     this.name = null 
    
    }
  );

  @override
  String toString() {
    return 'AssetCurrentLocationsResponse[cable=$cable, engineHours=$engineHours, id=$id, location=$location, name=$name, ]';
  }
}

@GenSerializer(nullableFields: true)
class AssetCurrentLocationsResponseSerializer extends Serializer<AssetCurrentLocationsResponse> with _$AssetCurrentLocationsResponseSerializer {

}

File: openapi-generator/scalaz/src/main/scala/org/openapitools/client/api/AssetCurrentLocationsResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import AssetCurrentLocationsResponse._

case class AssetCurrentLocationsResponse (
  /* The cable connected to the asset */
  cable: Option[List[AssetCable]],
/* Engine hours */
  engineHours: Option[Integer],
/* Asset ID */
  id: Long,
/* Current location of an asset */
  location: Option[List[AssetCurrentLocation]],
/* Asset name */
  name: Option[String])

object AssetCurrentLocationsResponse {
  import DateTimeCodecs._

  implicit val AssetCurrentLocationsResponseCodecJson: CodecJson[AssetCurrentLocationsResponse] = CodecJson.derive[AssetCurrentLocationsResponse]
  implicit val AssetCurrentLocationsResponseDecoder: EntityDecoder[AssetCurrentLocationsResponse] = jsonOf[AssetCurrentLocationsResponse]
  implicit val AssetCurrentLocationsResponseEncoder: EntityEncoder[AssetCurrentLocationsResponse] = jsonEncoderOf[AssetCurrentLocationsResponse]
}
